python_modules/dagster/dagster/core/execution/execute_in_process.py

```diff
@@ -63,6 +64,14 @@ def core_execute_in_process(
                 raise_on_error=raise_on_error,
             ),
         )
-        event_list = list(_execute_run_iterable)
+        event_list = list(execute_run_iterable)
 
-    return ExecuteInProcessResult(node, event_list, pipeline_run, output_capture)
+        is_success = all(not event.is_failure for event in event_list)
+
+        updated_pipeline_run = (
+            pipeline_run.with_status(DagsterRunStatus.SUCCESS)
+            if is_success
+            else pipeline_run.with_status(DagsterRunStatus.FAILURE)
+        )
```

Comment:
this being out of sync with the place computation where this happens means theres some risk for corner cases not to be handled correctly

I think thats probably fine for now but leave a comment here that this is emulating logic elsewhere in the system

Reply:
I used `handle_run_event` instead